### PR TITLE
redo forcepull test so localnode is not required

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://github.com/openshift/origin/issues/7101

@bparees PTAL

FYI - I toyed with the idea of just checking the pod pull flag for all three strategies but then just fell back to log scrapping for s2i/docker and checking the pod pull flag for custom.  I'll of course not protest changes along those lines.

### CODE REVIEW

**Comment by bparees** [comment]
checking the pod pull flag wouldn't work for the s2i and docker strategies because the pod pull flag has no bearing on the "base image" for the build in those strategies, it would only affect the pulling of the "openshift/sti-builder" and "openshift/docker-builder" images.

File: test/extended/builds/forcepull.go [hunk 1]
```diff
@@ -34,11 +30,32 @@ var (
...
+			fmt.Fprintf(g.GinkgoWriter, "\n\nfound pull image line %s\n\n", line)
+			found = true
+			break
+		}
+	}
 
-	if bldPrefix == buildPrefixFC || bldPrefix == buildPrefixTC {
+	if !found {
+		fmt.Fprintf(g.GinkgoWriter, "\n\n build log dump on failed test:  %s\n\n", out)
+		o.Expect(found).To(o.BeTrue())
```
Comment (bparees) [inline test/extended/builds/forcepull.go]: ExpectWithOffset?

File: test/extended/builds/forcepull.go [hunk 2]
```diff
@@ -79,17 +89,12 @@ we built.  The credentials also are recycled between those points.
...
 	g.BeforeEach(func() {
 
-		g.By("refresh corruptor, prep forcepull builder")
-		exutil.PullImage(corruptor, dockerClient.AuthConfiguration{})
-
-		exutil.DumpImage(corruptor)
-
 		// create the image streams and build configs for a test case specific builders
 		setupPath := exutil.FixturePath("testdata", "forcepull-setup.json")
 		err := exutil.CreateResource(setupPath, oc)
```
Comment (bparees) [inline test/extended/builds/forcepull.go]: is all this setup logic still needed, now that we aren't corrupting images?  (and thus don't need to worry about conflicting with other tests that are running on the node)?

**Comment by gabemontero** [comment]
Yeah it is needed since the BCs use DockerImage refs for the from images.
I drew the line at leaving that alone.  But sure I can switch the BCs to
use ImageTag and prune that set up stuff as well.

**Comment by gabemontero** [comment]
But if we keep them as DockerImage and put back in the explicit pull prior
to runnig the app build won't that remove the need of running the app build
twice.

**Comment by bparees** [comment]
> Yeah it is needed since the BCs use DockerImage refs for the from images.

yeah i don't see a reason to leave them that way w/ these changes, might as well make the imagestreams like all the other tests use.

> But if we keep them as DockerImage and put back in the explicit pull prior to runnig the app build

you can't do the explicit pull because it relies on the docker socket being available where the ginkgo framework is running.

**Comment by gabemontero** [comment]
On Tue, Dec 20, 2016 at 6:41 PM, Ben Parees <notifications@github.com>
wrote:

> Yeah it is needed since the BCs use DockerImage refs for the from images.
>
> yeah i don't see a reason to leave them that way w/ these changes, might
> as well make the imagestreams like all the other tests use.
>
> But if we keep them as DockerImage and put back in the explicit pull prior
> to runnig the app build
>
> you can't do the explicit pull because it relies on the docker socket
> being available where the ginkgo framework is running.
>
Yeah ... I realized that 5 minutes after suggesting it but wasn't at a
place where I could respond :-(

So net:

- switch to image streams; remove internal registry gorp
- trigger app builds twice to be better assured the pull occurred

**Comment by gabemontero** [comment]
changes pushed

File: test/extended/builds/forcepull.go [hunk 3]
```diff
@@ -79,18 +87,14 @@ we built.  The credentials also are recycled between those points.
...
 	g.BeforeEach(func() {
 
-		g.By("refresh corruptor, prep forcepull builder")
-		exutil.PullImage(corruptor, dockerClient.AuthConfiguration{})
-
-		exutil.DumpImage(corruptor)
-
 		// create the image streams and build configs for a test case specific builders
+		g.By("create image stream and builder build config")
 		setupPath := exutil.FixturePath("testdata", "forcepull-setup.json")
```
Comment (bparees) [inline test/extended/builds/forcepull.go]: i'm still not understanding why we need this instead of just using an existing s2i builder image from dockerhub

Comment (gabemontero) [inline test/extended/builds/forcepull.go]: Missed this nuance from the earlier exchange, but yeah, goes along with the fallout of no longer corrupting images.  I'll take a look.

Comment (gabemontero) [inline test/extended/builds/forcepull.go]: further removal pushed

**Comment by gabemontero** [comment]
[testextended][extended:core(forcePull)]

File: test/extended/builds/forcepull.go [hunk 4]
```diff
@@ -152,31 +100,31 @@ var _ = g.Describe("[LocalNode][builds] forcePull should affect pulling builder
...
-		g.It("ForcePull test case execution", func() {
-
-			g.By("when s2i force pull is false")
-
-			doTest(buildPrefixFS, "s2i false app/lang build", true, oc)
+		g.It("ForcePull test case execution s2i", func() {
 
 			g.By("when s2i force pull is true")
+			// run twice in case builder image was created on different node
+			// from the node processing this build
```
Comment (bparees) [inline test/extended/builds/forcepull.go]: "run twice to ensure the builder image gets pulled even if it already exists on the node"

File: test/extended/builds/forcepull.go [hunk 5]
```diff
@@ -152,31 +100,31 @@ var _ = g.Describe("[LocalNode][builds] forcePull should affect pulling builder
...
-
-			g.By("when docker force pull is false")
+		})
 
-			doTest(buildPrefixFD, "dock false app/lang build", true, oc)
+		g.It("ForcePull test case execution docker", func() {
 
 			g.By("docker when force pull is true")
+			// run twice in case builder image was created on different node
+			// from the node processing this build
```
Comment (bparees) [inline test/extended/builds/forcepull.go]: same here.

Comment (gabemontero) [inline test/extended/builds/forcepull.go]: sure thing ... will let the in progress extended test run complete before updating commit

**Comment by bparees** [comment]
[merge]

**Pull request merged**